### PR TITLE
Bedtools genomecov

### PR DESCRIPTION
**Pull request by tgaspe**
## Description

New component bedtools_genomecov

## Issue ticket number

Closes #xxxx 

## Checklist before requesting a review

- [x] I have performed a self-review of my code

- [x] Conforms to the [Contributing guidelines](https://github.com/viash-hub/base/blob/main/CONTRIBUTING.md)

- [x] Proposed changes are described in the [CHANGELOG.md](https://github.com/viash-hub/base/blob/main/CHANGELOG.md)

- [x] I have tested my code with `viash ns test --parallel -q <name or namespace>`

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [x] New functionality
  - [ ] Major changes
  - [ ] Minor changes
  - [ ] Documentation
  - [ ] Bug fixes

### CODE REVIEW

File: src/bedtools/bedtools_genomecov/config.vsh.yaml [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+          Calculate coverage of 5" positions (instead of entire interval).
+
+      - name: --three_prime
+        alternatives: -3
+        type: boolean_true
+        description: |
+          Calculate coverage of 3" positions (instead of entire interval).
+
+      - name: --max
+        type: integer
```
Comment (DriesSchaumont) [inline src/bedtools/bedtools_genomecov/config.vsh.yaml:147]:
```suggestion
        type: integer
        min: 0
```

File: src/bedtools/bedtools_genomecov/config.vsh.yaml [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        description: |
+          Combine all positions with a depth >= max into
+          a single bin in the histogram. Irrelevant
+          for -d and -bedGraph
+          - (INTEGER)
+
+      - name: --scale
+        type: double
```
Comment (DriesSchaumont) [inline src/bedtools/bedtools_genomecov/config.vsh.yaml:155]:
```suggestion
        type: double
        min: 0
```

File: src/bedtools/bedtools_genomecov/config.vsh.yaml [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+          Adds a UCSC/Genome-Browser track line definition in the first line of the output.
+          - See here for more details about track line definition:
+                http://genome.ucsc.edu/goldenPath/help/bedgraph.html
+          - NOTE: When adding a trackline definition, the output BedGraph can be easily
+                uploaded to the Genome Browser as a custom track,
+                BUT CAN NOT be converted into a BigWig file (w/o removing the first line).
+      
+      - name: --trackopts
+        type: string
+        description: |
```
Comment (DriesSchaumont) [inline src/bedtools/bedtools_genomecov/config.vsh.yaml:175]: Perhaps this is a good candidate for `multiple: true`

File: src/bedtools/bedtools_genomecov/script.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+[[ "$par_bed_graph" == "false" ]] && unset par_bed_graph
+[[ "$par_bed_graph_zero_coverage" == "false" ]] && unset par_bed_graph_zero_coverage
+[[ "$par_split" == "false" ]] && unset par_split
+[[ "$par_ignore_deletion" == "false" ]] && unset par_ignore_deletion
+[[ "$par_pair_end_coverage" == "false" ]] && unset par_pair_end_coverage
+[[ "$par_fragment_size" == "false" ]] && unset par_fragment_size
+[[ "$par_du" == "false" ]] && unset par_du
+[[ "$par_five_prime" == "false" ]] && unset par_five_prime
+[[ "$par_three_prime" == "false" ]] && unset par_three_prime
+[[ "$par_trackline" == "false" ]] && unset par_trackline
```
Comment (DriesSchaumont) [inline src/bedtools/bedtools_genomecov/script.sh:18]:
```suggestion
unset_if_false=(
  par_input_bam
  par_depth
  par_depth_zero
  par_bed_graph
  par_bed_graph_zero_coverage
  par_split
  par_ignore_deletion
  par_pair_end_coverage
  par_fragment_size
  par_du
  par_five_prime
  par_three_prime
  par_trackline
)

for par in ${unset_if_false[@]}; do
    test_val="${!par}"
    [[ "$test_val" == "false" ]] && unset $par
done
```

File: src/bedtools/bedtools_genomecov/script.sh [hunk 5]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+## VIASH START
+## VIASH END
+
```
Comment (DriesSchaumont) [inline src/bedtools/bedtools_genomecov/script.sh:5]: set -eo pipefail

File: src/bedtools/bedtools_genomecov/test.sh
Comment (DriesSchaumont) [inline src/bedtools/bedtools_genomecov/test.sh:1]: Could you write the test output to a temporary folder?

**Review by DriesSchaumont — CHANGES_REQUESTED** [review]

File: src/bedtools/bedtools_genomecov/config.vsh.yaml [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+        alternatives: -i
+        type: file
+        direction: input
+        description: |
+          The input file (BED/GFF/VCF) to be used.
+        example: input.bed
+      
+      - name: --input_bam
+        alternatives: -ibam
+        type: file
```
[the remark this answers is not in the document]
Comment (tgaspe) [reply src/bedtools/bedtools_genomecov/config.vsh.yaml:31]: That is what I thought at first as well, but it is not the case. 
It realy is a file type.

File: src/bedtools/bedtools_genomecov/config.vsh.yaml (hunk 3 above)
Comment (tgaspe) [reply src/bedtools/bedtools_genomecov/config.vsh.yaml:175]: I will look into it.

File: src/bedtools/bedtools_genomecov/config.vsh.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (DriesSchaumont) [reply src/bedtools/bedtools_genomecov/config.vsh.yaml:33]: You are right, my bad 👍

File: src/bedtools/bedtools_genomecov/config.vsh.yaml [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+  commands: [bedtools]
+authors:
+  - __merge__: /src/_authors/theodoro_gasperin.yaml
+    roles: [ author, maintainer ]
+
+argument_groups:
+  - name: Inputs
+    arguments:
+      - name: --input
+        alternatives: -i
```
[the remark this answers is not in the document]
Comment (DriesSchaumont) [reply src/bedtools/bedtools_genomecov/config.vsh.yaml:24]: Indeed!

File: src/bedtools/bedtools_genomecov/config.vsh.yaml (hunk 3 above)
Comment (tgaspe) [reply src/bedtools/bedtools_genomecov/config.vsh.yaml:179]: I have implemented multiple: true for this option

**Review by tverbeiren — APPROVED** [review]

**Pull request closed without merging**